### PR TITLE
[FIX] Work around a `get_json_object` libcudf bug, jsdom improvements

### PR DESCRIPTION
**Pull request by trxcllnt**

### CODE REVIEW

File: modules/jsdom/src/polyfills/events/keyboard.ts [hunk 1]
```diff
@@ -31,7 +31,13 @@ import {
 
 export function keyboardEvents(window: DOMWindow) {
   const keys        = keyUpdates(window);
-  const specialKeys = keys.pipe(filter(isSpecialKey));
+  const specialKeys = keys.pipe(filter(isSpecialKey)).pipe(flatMap(function*(keyEvt) {
```
Comment (thomcom) [inline modules/jsdom/src/polyfills/events/keyboard.ts:34]: What is the `*` for here in `function*`?

File: modules/jsdom/src/polyfills/events/keyboard.ts [hunk 2]
```diff
@@ -116,9 +129,10 @@ export class GLFWKeyboardEvent extends GLFWEvent {
     evt.target  = window;
     evt._repeat = action === glfw.REPEAT;
 
-    evt._key   = name || 'Unidentified';
-    evt._which = evt._charCode = glfwToDOMKey[key] || key;
-    evt._code                  = keyToCode[evt.which] || name || 'Unidentified';
+    evt._key      = glfwKeyToKey[key] || name || 'Unidentified';
+    evt._code     = glfwKeyToCode[key] || (name && `Key${name.toUpperCase()}`) || 'Unidentified';
+    evt._which    = glfwKeyToKeyCode[key] || key;
```
Comment (thomcom) [inline modules/jsdom/src/polyfills/events/keyboard.ts:134]: Love this usage of `||` to process variable inputs.

File: modules/jsdom/src/polyfills/events/mouse.ts [hunk 3]
```diff
@@ -37,19 +37,34 @@ export function mouseEvents(window: DOMWindow) {
...
 }
 
 function positionUpdates(window: DOMWindow) {
   return windowCallbackAsObservable(glfw.setCursorPosCallback, window)
-    .pipe(map(([, ...rest]) => GLFWMouseEvent.fromMouseMove(window, ...rest)))
+    .pipe(flatMap(function*([, ...rest]) {
+      const mouseEvt = GLFWMouseEvent.fromMouseMove(window, ...rest);
+      yield mouseEvt;
+      yield Object.assign(GLFWMouseEvent.fromMouseMove(window, ...rest),
+                          {type: `pointer${mouseEvt.type.slice(5)}`});
```
Comment (thomcom) [inline modules/jsdom/src/polyfills/events/mouse.ts:55]: Similar to `function*`, what is `pointer$` here?

**Review by thomcom — APPROVED** [review]

File: modules/jsdom/src/polyfills/events/keyboard.ts (hunk 1 above)
Comment (trxcllnt) [reply modules/jsdom/src/polyfills/events/keyboard.ts:34]: That indicates this a [generator function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator).

File: modules/jsdom/src/polyfills/events/mouse.ts (hunk 3 above)
Comment (trxcllnt) [reply modules/jsdom/src/polyfills/events/mouse.ts:55]: this is making event type from the current `mouse<blah>` to `pointer<blah>`.

Comment (thomcom) [reply modules/jsdom/src/polyfills/events/mouse.ts:55]: Oh duh it is just a `python f string`.

**Pull request merged**